hare/algorithm.go

```diff
@@ -80,7 +80,6 @@ type State struct {
 // Messages are sent as type Message. Upon receiving, the NodeID is added to this wrapper (public key extraction).
 type Msg struct {
 	Message
-	NodeID types.NodeID
 }
```

Comment:
This could probably be removed, but constructing a `Msg` from a `Message` seems to do so much more than just extracting the `NodeID` that I didn't want to pollute this PR with removing this type.